builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -186,6 +195,7 @@ func resourceCloudStackInstanceRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("display_name", vm.Displayname)
 	d.Set("ipaddress", vm.Nic[0].Ipaddress)
 	d.Set("zone", vm.Zonename)
+	//NB cloudstack sometimes sends back the wrong keypair name, so dont update it
```

Comment:
OK cool I will raise that then. 

About reading it back, I'm not really sure I'd say it is safe for the way I'm using this at the moment (non-converging keypair values causes a load of unneccessary reboots on every update), but I agree it seems like the best option in general. 

So, I must test if I can avoid this reboot-on-update issue by de-duplicating my SSH keys. In the meantime I will make the change and update this PR